### ISSUE TITLE
OMSAgent segfaults every 16s on RHEL

### ISSUE THREAD

**Issue by pentiumoverdrive** [microsoft/OMS-Agent-for-Linux#1193]
I'm running OMSAgent on RHEL 7.4 and the OMSAgent (OMI) keeps segfaulting about every 16s.

I checked the OMI repo and they had some tips on upgrading packages, which I did. No difference. It kept filling up the disk with coredumps eventually.

This works on loads of other systems so I'm not sure what I am running into here.

**rpm -qa|grep nss-pem**
nss-pem-1.0.3-7.el7.x86_64

**rpm -qa|grep curl**
python-pycurl-7.19.0-19.el7.x86_64
curl-7.29.0-54.el7.x86_64
libcurl-7.29.0-54.el7.x86_64

**rpm -qa|grep oms**
omsconfig-1.1.1-927.x86_64
omsagent-1.12.15-0.x86_64
auoms-2.0.1-19.x86_64

**Error message:**
> 
> May 18 08:59:12 server01 kernel: omiagent[23718]: segfault at 0 ip 00007f993ea420ff sp 00007fff3869e720 error 4 in libSCXCoreProviderModule.so[7f993e888000+342000]
> May 18 08:59:28 server01 kernel: omiagent[23889]: segfault at 0 ip 00007f0f697300ff sp 00007ffe0a62cba0 error 4 in libSCXCoreProviderModule.so[7f0f69576000+342000]
> May 18 08:59:44 server01 kernel: omiagent[24055]: segfault at 0 ip 00007f8e8c1c50ff sp 00007ffe47de4f90 error 4 in libSCXCoreProviderModule.so[7f8e8c00b000+342000]
> May 18 09:00:00 server01 kernel: omiagent[24133]: segfault at 0 ip 00007f2f6a2c90ff sp 00007ffd962ee190 error 4 in libSCXCoreProviderModule.so[7f2f6a10f000+342000]
> May 18 09:00:17 server01 kernel: omiagent[24230]: segfault at 0 ip 00007fd517b510ff sp 00007ffd78156550 error 4 in libSCXCoreProviderModule.so[7fd517997000+342000]

**Comment by pentiumoverdrive** [comment]
the omsagent.log throws one gigantic long line starting with:

2020-05-18 08:17:06 +0200 [warn]: dump an error event: error_class=NoMethodError error="undefined method `+' for nil:NilClass" tag="oms.patch_management_immediate_run" time=1589782626 record={"xml"=>"

**Comment by pentiumoverdrive** [comment]
Guess what, I solved it. This took many hours :(
I found another system that had the same problem (test system) so I started poking around some.

Eventually when installing I noticed:

"----- Upgrading package: scx (scx-1.6.3-1079.universal.x64) -----                                                                                                                                
Unable to convert default host name ''.                                                                                                                                                          
This might be caused by a host name that contains UTF-8 characters that are invalid given your current locale. "

After logs of digging around in host files etc I did this:

>  hostnamectl --pretty set-hostname <server-fqdn>

..and suddenly the installer could generate the correct certificate and no more errors.

I even tried using: **/opt/microsoft/scx/bin/tools/scxsslconfig -f -h hostname -d domain -v** but it would just throw a "**CXCoreLib::SCXStringConversionException**" and nothing more. Pretty bad.

Maybe add some improvements to the preflight checks!
